File: java/1903. Largest Odd Number in String.java
/**
 *Problem: 1903. Largest Odd Number in String
 *You are given a string num, representing a large integer. Return the largest-valued odd integer (as a string) that is a non-empty substring of num, or an empty string "" if no odd integer exists.
 *A substring is a contiguous sequence of characters within a string.
 */

class Solution {
    //Approach: Find the Rightmost Odd Digit
    public String largestOddNumber(String num) {
        for(int i = num.length()-1; i>=0; i--){
            if(Character.getNumericValue(num.charAt(i))%2!=0){
                return num.substring(0,i+1);
            }
        }
        return "";
    }
}

File: python3/1903. Largest Odd Number in String.py
'''
Problem: 1903. Largest Odd Number in String
You are given a string num, representing a large integer. Return the largest-valued odd integer (as a string) that is a non-empty substring of num, or an empty string "" if no odd integer exists.
A substring is a contiguous sequence of characters within a string.
'''

class Solution:
    def largestOddNumber(self, num: str) -> str:
      #range(starting point, ending point but not included, step value)
      #len(num)-1 is the index of the last character in the string 'num'
      for i in range(len(num) -1,-1,-1):
        if int(num[i])%2!=0:
          #num: The original string.
          #[:i+1]: The slicing syntax.
          #i+1: The ending index for the slice.
          return num[:i+1]

      return ""
        
